download/download.go

```diff
@@ -276,6 +276,8 @@ func (d *Downloader) download(ctx context.Context, m metrics.Metrics) (*download
 			t := int64(0)
 			d.client = d.client.SetResponseHeaderTimeout(&t)
 		}
+	} else {
+		d.client = d.client.WithoutHeader("Prefer")
 	}
```

Comment:
I am not sure why need to explicitly remove the header. The server will probably ignore it if it does not support long polling.

Reply:
It's more about consistency (or validity ?) than about server.  

* When `d.longPollingEnabled` is true, each time `download` function is invoked, `prefer` header is put into client, and client is set to download object.
* In point of view of state management, it's (for me) important to handle the reverse side effect when `d.longPollingEnabled` switch to false. It's avoid inconsistency.

Notice:

I don't understand why it's added and save each time `download` function is invoked: (a) if it's save, don't need to re-add, (b) if not save, need to re-add. I suppose/imagine it's for avoid add `if prefer is already present`. I wonder if it work to set this header in `New` function. In this case,  it's not depend on longPollingEnabled (at runtime) but only on configuration.

Reply:
In that case you can set the header to have a `wait` preference of 0. Then you don't need to add a new public method.

Reply:
Done.